File: pinyougou-interface/src/main/java/cn/itcast/core/service/SeckillGoodsService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.seckill.SeckillGoods;
import cn.itcast.core.pojo.item.Item;
import cn.itcast.core.pojo.seckill.SeckillGoods;
import entity.PageResult;

import java.util.List;

public interface SeckillGoodsService {
    PageResult search(Integer page, Integer rows, SeckillGoods seckillGoods);

    List<Item> findItemList(String sellerId);

    Item findByItem(Long itemId);

    void save(SeckillGoods seckillGoods);

    public List<SeckillGoods> findList();
    /**
     * 根据ID获取实体(从缓存中读取)
     */
    public SeckillGoods findOneFromRedis(Long id);



}

File: pinyougou-web-shop/src/main/webapp/js/service/seckillGoodsService.js
app.service('seckillGoodsService',function ($http) {

    this.search=function (page,rows,searchEntity) {

        return $http.post("../seckillGoods/search.do?page="+page+"&rows="+rows,searchEntity);
    }

    //根据id查询item信息
    this.findByItem=function (itemId) {
        return $http.get("../seckillGoods/findByItem.do?itemId="+itemId);
    }

    this.findItemList=function () {
        return $http.get("../seckillGoods/findItemList.do");

    }

    this.save=function (seckillGoods) {
        return $http.post("../seckillGoods/save.do",seckillGoods);

    }


})